neutron/diffraction/convert_with_calibration.cpp

```diff
@@ -71,11 +71,20 @@ template <class T> T convert_with_calibration_impl(T d, Dataset cal) {
   return d;
 }
 
+void validate_calibration(const Dataset &cal) {
+  if (cal["tzero"].unit() != scipp::units::us)
+    throw except::UnitError("tzero must have units of `us`");
+  if (cal["difc"].unit() != scipp::units::us / scipp::units::angstrom)
+    throw except::UnitError("difc must have units of `us / angstrom`");
```

Comment:
Will break if someone starts to use, e.g., `ms` as their unit. Not sure how likely that is, so maybe we can ignore that.